v2client/common/common.go

```diff
@@ -0,0 +1,211 @@
+package common
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/google/go-querystring/query"
+)
+
+const (
+	authHeader           = "X-Algo-API-Token"
+	healthCheckEndpoint  = "/health"
+	apiVersionPathPrefix = "/v2"
+)
+
+// unversionedPaths ais a set of paths that should not be prefixed by the API version
+var unversionedPaths = map[string]bool{
+	"/versions": true,
+	"/health":   true,
+}
+
+// rawRequestPaths is a set of paths where the body should not be urlencoded
+var rawRequestPaths = map[string]bool{
+	"/transactions": true,
+}
+
```

Comment:
@winder I wasn't able to understand from the spec, should `transactions` still be a rawRequestPath? See L128 below